### PR TITLE
Add post:JS_object_compare

### PR DESCRIPTION
**Pull request by yiyu0x**

### CODE REVIEW

**Comment by kdchang** [comment]
Thank you @yiyu0x for your submission. :)

hi @ArvinH  @aszx87410 @Po-Jen , please help to review this PR. Thank you :)

**Comment by kdchang** [comment]
Hi @yiyu0x,

Sorry for the late reply. Thank you for your sharing! Would you mind adding some recap in your conclusion and add some author introduction in the last of the post ? Thank you :)

cc @ArvinH @aszx87410 @Po-Jen

**Comment by yiyu0x** [comment]
Hello @kdchang  ,

Thanks for your reply and suggestion. 
I add "about me" paragraph and more recap in my post :)

File: source/_posts/javascript-obj-comare.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+```
+let str = 'hi'
+let str2 = String('hi')
+```
+
+這兩個字串的內容到底一樣不一樣？他們都是 primitive data type，所以當然一樣。如果用 `new String('hi')` 才會不一樣！
+
+理解這些平常不會仔細思考的問題更有助於掌握這門語言，日後我也會時常將我的學習歷程寫成文章，發佈於我的部落格與此，謝謝大家仔細看完這篇文章。
+
+## 關於我
```
Comment (kdchang) [inline source/_posts/javascript-obj-comare.md]: suggest update
## 關於我 
-> 關於作者

**Comment by kdchang** [comment]
Hi @yiyu0x ,

Thank you :) It seems that the files changed scope cover too many files ? 
Have you fetched the latest upstream version to your local and done your branch git rebase from upstream/master ?

**Comment by yiyu0x** [comment]
Hi @kdchang , 

I forgot to fetch the latest upstream version yesterday, and I update something in new commits.
Thanks for checking, have a good day:)

File: source/_posts/javascript-obj-compare.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
+---
+title: 理解 Javascript 中物件的比大小行為
```
Comment (aszx87410) [inline source/_posts/javascript-obj-compare.md]: 話說這個字可以修一下，正確寫法是 JavaScript

File: source/_posts/javascript-obj-compare.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Javascript 的型態簡單來說就如同前面所說，一類為 primitive data type 另外一類就是物件(object)。很多新手常常搞混的行為其實也就只是這兩種屬性不太了解罷了。
+
+像是:
+
+```javascript
+let str = 'hi'
+let str2 = String('hi')
+```
+
+這兩個字串的內容到底一樣不一樣？他們都是 primitive data type，所以當然一樣。如果用 `new String('hi')` 才會不一樣！
```
Comment (aszx87410) [inline source/_posts/javascript-obj-compare.md]: 這邊我覺得可以再補充然後講細一點。

因為如果你要說「內容」一不一樣的話，`new String('hi')` 跟 `'hi'` 我也會認為是一樣的，因為 `new String('hi') == 'hi'` 會是 true，但我知道你想表達的是前者是一個 Object，後者是 primitive type，所以他們的 type 其實是不一樣的。

Comment (yiyu0x) [reply source/_posts/javascript-obj-compare.md]: 了解，想說這不是這篇主要探討的問題，講太多會失焦。
但補充詳細一點當然也沒問題。

File: source/_posts/javascript-obj-compare.md (hunk 2 above)
Comment (yiyu0x) [reply source/_posts/javascript-obj-compare.md]: 感謝告知，已修正。

**Comment by yiyu0x** [comment]
Hi @aszx87410 ,

Thanks for your suggestions. Please recheck :)

**Review by aszx87410 — APPROVED** [review]

**Review by kdchang — APPROVED** [review]
LGTM, we will release this post in this weekend newsletter. Thank you @yiyu0x :)

**Pull request merged**